Sources/MessagingPush/MessagingPushConfigOptions.swift

```diff
@@ -62,4 +62,12 @@ public struct MessagingPushConfigOptions {
      operating system, device locale, device model, app version etc
      */
     public var autoTrackDeviceAttributes: Bool
+
+    /// Applies new configuration to this object
+    /// - Important: Ensure all mutable properties are updated accordingly
+    mutating func apply(_ newConfig: MessagingPushConfigOptions) {
+        autoFetchDeviceToken = newConfig.autoFetchDeviceToken
+        autoTrackPushEvents = newConfig.autoTrackPushEvents
+        autoTrackDeviceAttributes = newConfig.autoTrackDeviceAttributes
+    }
```

Comment:
For safely updating config without updating reference

Reply:
Based on recent PR conversations we have had, instead of a mutating function, could we leverage constructors again? 

Here is a code sample that is similar to what is in this file:

```swift
public static func create(from newConfig: MessagingPushConfigOptions) -> MessagingPushConfigOptions {
            return MessagingPushConfigOptions(
                autoFetchDeviceToken: newConfig.autoFetchDeviceToken
                ...
            )
}
```

Reply:
I added this for `configure` method only, which is why I avoided recreating the instance. Since we're already [discussing removing it](https://github.com/customerio/customerio-ios/pull/416#discussion_r1408136656), once we finalize that decision, I'll also remove this method with that change.